### PR TITLE
Add support for python 3.7 in travis ci

### PR DESCRIPTION
**Pull request by sushobhana**
#206 
The sudo and dist must be `true` and `xenial` explicitely for python3.7 (see https://github.com/travis-ci/travis-ci/issues/9815)

### CODE REVIEW

**Comment by cdeil** [comment]
There's one issue with the Sphinx install here:
https://travis-ci.org/astropy/regions/jobs/418661367#L946

That looks unrelated to our package. @sushobhana - unless you know how to fix or work around that issue, or want to engage with pip or Sphinx to fix it upstream, you could just keep the docs build / test on Python 3.6 for now, and then this can go in.

Could you try to change one Mac and / or Windows build to 3.7 as well?

File: .travis.yml [hunk 1]
```diff
@@ -39,42 +39,51 @@ matrix:
     include:
 
         # Do a coverage test in Python 3.
-        - python: 3.6
+        - python: 3.7
```
Comment (bsipocz) [inline .travis.yml]: with other packages we leave it to conda to deal with the python versions (via the `PYTHON_VERSION` variable) and use `generic` or `c` as the travis language. Maybe that would solve your issue here, too?

**Comment by larrybradley** [comment]
Re: `regions 0.2` not being able to install via `pip` with `Python 3.7`:

This is a known issue with Cython and Python 3.7.  Basically, the distribution currently on PyPI was built with an older version of Cython (and that version is incompatible with Python 3.7).  If it gets rebuilt with a newer Cython it should work.

**Comment by larrybradley** [comment]
As a workaround, you should be able to install directly from the github tarball.

**Comment by sushobhana** [comment]
Thanks @bsipocz and @larrybradley . @cdeil - Added support for windows and osx as well.

**Comment by cdeil** [comment]
@sushobhana - Thanks for setting this up, and @mwcraig - thanks for the advice!

**Pull request merged**